soc/arm/nordic_nrf/nrf53/soc.c

```diff
@@ -120,6 +130,19 @@ static int nordicsemi_nrf53_init(const struct device *arg)
 	nrf_regulators_dcdcen_vddh_set(NRF_REGULATORS, true);
 #endif
 
+#if defined(CONFIG_SOC_NRF5340_CPUAPP) && \
+	DT_HAS_COMPAT_STATUS_OKAY(nordic_nrf_gpio_forwarder)
+	static const struct gpio_dt_spec pf_gpios[] = {
+		DT_FOREACH_CHILD(DT_INST(0, nordic_nrf_gpio_forwarder), ALL_GPIOS_IN_NODE)
+	};
+
+	for (int i = 0; i < ARRAY_SIZE(pf_gpios); i++) {
+		soc_secure_gpio_pin_mcu_select(pf_gpios[i].pin,
```

Comment:
This will work only for `gpio0`, since this function needs a PSEL value (a pin number combined with the related GPIO port number on bit 5) and `pf_gpios[i].pin` is a pin number within a given GPIO port.
You'd rather need to use an array of PSEL values. I think that `soc_nrf_common.h` should be extended with a macro similar to [`NRF_DT_GPIOS_TO_PSEL()`](https://github.com/zephyrproject-rtos/zephyr/blob/52be66919ec454e0943292be325b1fdb8f3ef2ec/soc/arm/nordic_nrf/common/soc_nrf_common.h#L129-L131) but taking additionally the index parameter, and then this macro should be used above instead of `GPIO_DT_SPEC_GET_BY_IDX()`.

Reply:
I added `NRF_DT_GPIOS_TO_PSEL_BY_IDX` as you suggested, please have a look